### PR TITLE
Add build:watch script

### LINKED ISSUE #1089: monorepo build:watch
**Is your feature request related to a problem? Please describe.**
It's frustrating when eg: Running the cli from `lodestar-cli`, after updating code in `lodestar`, we must run `yarn build` from the lodestar package, for `lodestar-cli` to pick up the changes.

The typescript source is not used by other downstream monorepo packages, only the transpiled javascript is used.

**Describe the solution you'd like**
Maybe we can have something along the lines of `babel --watch && tsc --watch` running in each monorepo package, which will pick up any typescript file changes and automatically update the transpiled JS. This way, we won't need to be constantly be manually re-building each package.
If we can package this into something that we can run via `lerna`, we can get auto-building for the whole monorepo at once, very nice.

**Describe alternatives you've considered**
write a custom thing with https://github.com/paulmillr/chokidar with a nice glob that targets only what we need

**Additional context**
https://babeljs.io/docs/en/babel-cli#compile-files
https://www.typescriptlang.org/docs/handbook/configuring-watch.html

### PR DESCRIPTION
**Pull request by dapplion**
Fixes https://github.com/ChainSafe/lodestar/issues/1089.

- Adds build watch scripts to all packages. Devs can target the packages in development with lerna's scope.
- Otherwise, it also adds a concurrent script in the root package.json so everything is watched at once. Can be heavy on memory ~4GB on my laptop.

### CODE REVIEW

File: packages/lodestar-utils/src/assert.ts [hunk 1]
```diff
@@ -20,6 +20,9 @@ export const assert = {
     }
   },
 
+  hello: 2,
+  bye: "a",
```
Comment (mpetrunic) [inline packages/lodestar-utils/src/assert.ts]: eh? :smile:

**Review by mpetrunic — CHANGES_REQUESTED** [review]
LGTM just some junk left

File: packages/lodestar-utils/src/assert.ts (hunk 1 above)
Comment (dapplion) [reply packages/lodestar-utils/src/assert.ts]: Thanks for catching that!

File: packages/lodestar-cli/src/index.ts [hunk 2]
```diff
@@ -7,6 +7,10 @@ import {validatorCommandModule} from "./cmds/validator";
 import {globalOptions} from "./options";
 import {YargsError} from "./util/errors";
 
+import {assert} from "@chainsafe/lodestar-utils";
+
+assert.hello;
+assert.bye+1;
```
Comment (mpetrunic) [inline packages/lodestar-cli/src/index.ts]: ehem :rofl:

**Review by mpetrunic — CHANGES_REQUESTED** [review]

File: packages/lodestar-cli/src/index.ts (hunk 2 above)
Comment (dapplion) [reply packages/lodestar-cli/src/index.ts]: hahaha :shushing_face:

**Review by mpetrunic — APPROVED** [review]

**Review by wemeetagain — APPROVED** [review]

**Pull request merged**